### ISSUE TITLE
Incorrect type hint for std.os.windows.PathSpace.span()

### ISSUE THREAD

**Issue by JustinBraben** [zigtools/zls#2082]
### Zig Version

0.13.0

### ZLS Version

0.13.0

### Client / Code Editor / Extensions

windows, vscode with zsl

### Steps to Reproduce and Observed Behavior

Create this file:

```zig
const std = @import("std");

pub fn main() !void {
    const file_name = "fmtest.txt";
    const dir = std.fs.cwd();
    const path_w = try std.os.windows.sliceToPrefixedFileW(dir.fd, file_name);
    const lpcTheFile = path_w.span();

    _ = lpcTheFile;
}
```

Gives []const u16 type hint
![image](https://github.com/user-attachments/assets/0e95cc71-5000-4467-b126-82228071f83a)
Also []const u16 in the hover type hint
![image](https://github.com/user-attachments/assets/18b8b989-eeba-4767-b2ed-444cdaaabe42)

### Expected Behavior

Both hints should give [:0]const u16 type hint, as denoted in `std.os.windows.PathSpace.span()`

**Comment by JustinBraben** [comment]
Out of curiosity tried running the former code on the latest zls [6e0e7a8
](https://github.com/zigtools/zls/commit/6144e778307f65c18308bb12fa5a25add91afb27) and it does get the correct [:0]const u16 type hint. So this can be closed.

Also tried running this: 
```zig
test "function with output sentinel" {
    const code = 
        \\const std: type = @import("std");
        \\const file_name: *const [10:0]u8 = "fmtest.txt";
        \\const dir: Dir = std.fs.cwd();
        \\const path_w: PathSpace = try std.os.windows.sliceToPrefixedFileW(dir.fd, file_name);
        \\const lpcTheFile_1: []const u16 = path_w.span();
        \\const lpcTheFile_2: [:0]const u16 = path_w.span();
        \\_ = lpcTheFile_1;
        \\_ = lpcTheFile_2;
    ;
    try testInlayHints(code, .{ .kind = .Type });
}
```

Which actually passes, and left me wondering are []const T and [:0]const T considered the same in zig?

**Comment by Techatrix** [comment]
> are []const T and [:0]const T considered the same in zig?

No, ZLS was just not resolving the sentinel value. This was fixed by #1920.

The inlay hint test you have written doesn't test anything.
